kedro/runner/runner.py

```diff
@@ -224,6 +216,19 @@ def _suggest_resume_scenario(
                 f"argument to your previous command:\n{postfix}"
             )
 
+    @staticmethod
+    def _release_datasets(
+        node: Node, catalog: CatalogProtocol, load_counts: dict, pipeline: Pipeline
+    ) -> None:
+        """Decrement dataset load counts and release any datasets we've finished with"""
+        for dataset in node.inputs:
+            load_counts[dataset] -= 1
+            if load_counts[dataset] < 1 and dataset not in pipeline.inputs():
```

Comment:
```suggestion
            if load_counts[dataset] and dataset not in pipeline.inputs():
```

Reply:
@ElenaKhaustova can you explain the suggestion of removing the check for the `load_counts[dataset]` to be smaller than 1?

Reply:
^I think the idea is if `load_counts[dataset]==0`, it is evaluated as `False` so the condition `<1` isn't strictly needed. 

In this case I prefer to keep `load_counts[dataset] <1` since semantically we are doing reference counting.